blockchain/accept.go

```diff
@@ -84,9 +84,11 @@ func (b *BlockChain) maybeAcceptBlock(block *btcutil.Block, flags BehaviorFlags)
 	// Notify the caller that the new block was accepted into the block
 	// chain.  The caller would typically want to react by relaying the
 	// inventory to other peers.
-	b.chainLock.Unlock()
-	b.sendNotification(NTBlockAccepted, block)
-	b.chainLock.Lock()
+	func() {
```

Comment:
What exactly does the function closure achieve here? Can't we just do the plain defer?

Reply:
Is it attempting to rely on defer order execution after a panic?